### PR TITLE
llm-fundamental - Langchain 0.1 updates

### PR DESCRIPTION
**Pull request by martinohanlon**
Updates to reflect the changes made in Langchain 0.1

### CODE REVIEW

File: asciidoc/courses/llm-fundamentals/modules/3-intro-to-langchain/lessons/7-c-neo4j-vector-retriever/code/retriever_agent.py [hunk 1]
```diff
@@ -27,61 +23,67 @@
...
+    results = plot_retriever.invoke({"query":query})
+    # format the results
+    movies = '\n'.join([doc.metadata["title"] + " - " + doc.page_content for doc in results["source_documents"]])
+    return movies
 
 tools = [
     Tool.from_function(
-        name="ChatOpenAI",
-        description="For when you need to chat about movies, genres or plots. The question will be a string. Return a string.",
+        name="Movie Chat",
```
Comment (adam-cowley) [inline asciidoc/courses/llm-fundamentals/modules/3-intro-to-langchain/lessons/7-c-neo4j-vector-retriever/code/retriever_agent.py:57]: Do these need to be UpperCamelCase?

Comment (martinohanlon) [reply asciidoc/courses/llm-fundamentals/modules/3-intro-to-langchain/lessons/7-c-neo4j-vector-retriever/code/retriever_agent.py:57]: Dont think so. They work fine with textual descriptions.

One of the changes I found in v 0.1 was unless I had "better" names for the tools the llm was less likely to pick the right tool based on the query.

Comment (martinohanlon) [reply asciidoc/courses/llm-fundamentals/modules/3-intro-to-langchain/lessons/7-c-neo4j-vector-retriever/code/retriever_agent.py:57]: Thought I should check whether it was a convention to use UpperCamelCase. I dont think it is, the langchain documents have examples in "normal" text, CamelCase, kebab-case.

**Comment by martinohanlon** [comment]
This PR is ready to go.

**Review by adam-cowley — APPROVED** [review]

**Pull request merged**